pvdumpApp/src/pvdump.cpp

```diff
@@ -218,40 +220,27 @@ static sql::Driver* mysql_driver = NULL;
 
 static const int MAX_MACRO_VAL_LENGTH = 100; // should agree with length of macroval in iocenv MySQL table (iocdb_mysql_schema.txt)
 
-static int dumpMysql(const std::map<std::string,PVInfo>& pv_map, int pid, const std::string& exepath)
+struct MysqlThreadArgs
+{
+    const std::map<std::string,PVInfo>& pvm;
+    const std::list<std::string>& evl;
```

Comment:
would it be better to use a vector here as we're not doing insertion?

Reply:
Vectors need to be sized appropriately or else you get a performance penalty when they need to resize and copy, so rather than walk the `environ` pointer list an extra time to get the vector size I just went for a c++ list as performance wasn't critical for this bit.

Reply:
Given data is only appended `std::forward_list` would be more efficient on storage, but not sure it is any quicker